node/pkg/watchers/evm/verify_chain_config/verify.go

```diff
@@ -0,0 +1,88 @@
+package main
+
+// This is a tool that queries the RPCs to verify that the `EvmChainID` values in the chain config maps are correct as
+// compared to the specified well-known public RPC endpoint. This tool should be run whenever either of the chain config maps are updated.
+
+// Usage: go run verify.go [--env `env``]
+//        Where `env` may be "mainnet", "testnet" or "both" where the default is "both".
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"sort"
+
+	"github.com/certusone/wormhole/node/pkg/common"
+	"github.com/certusone/wormhole/node/pkg/watchers/evm"
+	"github.com/wormhole-foundation/wormhole/sdk/vaa"
+)
+
+var (
+	envStr = flag.String("env", "both", `Environment to be validated, may be "mainnet", "testnet" or "both", default is "both"`)
+)
+
+func main() {
+	flag.Parse()
+
+	if *envStr == "both" {
+		verifyForEnv(common.MainNet)
+		verifyForEnv(common.TestNet)
+	} else {
+		env, err := common.ParseEnvironment(*envStr)
+		if err != nil || (env != common.TestNet && env != common.MainNet) {
+			if *envStr == "" {
+				fmt.Printf("Please specify --env\n")
+			} else {
+				fmt.Printf("Invalid value for --env, should be `mainnet`, `testnet` or `both`, is `%s`\n", *envStr)
+			}
+			os.Exit(1)
+		}
+
+		verifyForEnv(env)
+	}
+}
+
+type ListEntry struct {
+	ChainID vaa.ChainID
+	Entry   evm.EnvEntry
+}
+
+func verifyForEnv(env common.Environment) {
+	m, err := evm.GetChainConfigMap(env)
+	if err != nil {
+		fmt.Printf("Failed to get chain config map for %snet\n", env)
+		os.Exit(1)
+	}
+
+	// The map is ordered by hash. Sort it by chain ID.
+	orderedList := []ListEntry{}
```

Comment:
Just as a fun fact (i.e. please ignore if you want 😄 ), you can do this:

```
orderedList := make([]ListEntry, len(m))
i := 0
for chainId, entry := range m {
    orderedList[i] = ListEntry{chainId: chainId, entry: entry}
    i++
}
```
There can be performance benefits with this pattern because `append()` does a copy and re-allocation under-the-hood as the slice exceeds it's previous `capacity`. On the other hand if you know the length of the slice in advance, you can avoid those intermediate re allocs.

Of course, this map is very small and this is a standalone tool so this isn't really relevant here, just wanted to share.

Reply:
I agree but this is is just a "quick and dirty" tool. 😁